### ISSUE TITLE
Release 4.1.1

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1224]
Can be labelled **ready** when all other issues on this milestone are closed.

- [ ] ~~run code analysis in VS in *Release* mode and address all coding and style violations (send a regular PR which must be merged before continuing)~~
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- [x] run `prepare_release.cmd x.y.z` to:
    - change "vNext" names to appropriate x.y.z version for the next GitHub Release, milestone, and release checklist (this issue)
    - create a new draft GitHub Release, milestone, and release checklist
    - create release/x.y.z branch on upstream, branching from develop
    - submit a PR to merge the release/x.y.z branch to master
- [x] once release-x.y.z has been merged, tag master and push to upstream. This will push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy) and attach all nupkgs from the tagged build to the GitHub Release.
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [x] close this issue
- [ ] close the [milestone](https://github.com/FakeItEasy/FakeItEasy/milestones?state=open) for this release

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/917791788223434752
https://twitter.com/fakeiteasyfx/status/917791788223434752

**Comment by blairconrad** [comment]
Based on several reports (#1230, #1239) of our bad error reporting when the wrong type is used in a constraint (and it seems like that's popular activity!) I think we should release 4.1.1 real soon now.
In fact, I think we're ready to go with the two issues we have committed. I'd happily remove #1228 from the milestone (and if we get a successful release) close it.
Sadly, I'm away from a computer for the rest of the day, but could initiate a release tomorrow morning.

**Comment by thomaslevesque** [comment]
> I think we should release 4.1.1 real soon now.
> In fact, I think we're ready to go with the two issues we have committed

I agree, but I was under the impression that you wanted to include wrong-typed argument constraint detection in that release.

> Sadly, I'm away from a computer for the rest of the day, but could initiate a release tomorrow morning.

I should be able to do it tonight.

**Comment by thomaslevesque** [comment]
Ah, I always forget that I can't merge the release PR myself... It's a bit annoying that the release process requires the intervention of two owners. I wonder if we shouldn't have a "bot" account, used by the prepare_release script, so that we could merge the PR created by the script.

**Comment by blairconrad** [comment]
> I always forget that I can't merge the release PR myself...

If only you'd let the script just commit and merge itself! :wink:

> It's a bit annoying that the release process requires the intervention of two owners. 

True. It is nice to have extra eyes on the commit, but mostly not necessary.

>  I wonder if we shouldn't have a "bot" account, used by the prepare_release script, so that we could merge the PR created by the script.

Intriguing. I'm not against it. I guess the organization can't issue tokens?

>  I was under the impression that you wanted to include wrong-typed argument constraint detection in that release

It'd've been nice, but since we're having complaints about the bug, I don't mind waiting. Heck. I don't mind releasing 4.2 a day after 4.1.1 if it comes to that. The release process is getting ever lighter!

**Comment by thomaslevesque** [comment]
> If only you'd let the script just commit and merge itself! 😉

Ah! You got me there.

> I guess the organization can't issue tokens?

I don't think it can.

**Comment by thomaslevesque** [comment]
NuGet down?

```
Pushing FakeItEasy.4.1.1.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  InternalServerError https://www.nuget.org/api/v2/package/ 41712ms
  PUT https://www.nuget.org/api/v2/package/
  InternalServerError https://www.nuget.org/api/v2/package/ 40882ms
  PUT https://www.nuget.org/api/v2/package/
```

**Comment by thomaslevesque** [comment]
https://twitter.com/fakeiteasyfx/status/920795511564709889
